crates/test-cluster/src/indexer_util.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use jsonrpsee::http_client::{HttpClient, HttpClientBuilder};
+use std::path::PathBuf;
+use std::time::Duration;
+use sui_config::local_ip_utils::get_available_port;
+use sui_indexer::tempdb::TempDb;
+use sui_indexer::test_utils::{
+    start_indexer_jsonrpc_for_testing, start_indexer_writer_for_testing,
+};
+use sui_json_rpc_api::ReadApiClient;
+use sui_sdk::{SuiClient, SuiClientBuilder};
+use tempfile::TempDir;
+use tokio::time::sleep;
+
+pub(crate) struct IndexerHandle {
+    pub(crate) rpc_client: HttpClient,
+    pub(crate) sui_client: SuiClient,
+    pub(crate) rpc_url: String,
+    #[allow(unused)]
+    cancellation_tokens: Vec<tokio_util::sync::DropGuard>,
+    #[allow(unused)]
+    data_ingestion_dir: Option<TempDir>,
+    #[allow(unused)]
+    database: TempDb,
+}
+
+// TODO: this only starts indexer writer and reader (jsonrpc server) today.
+// Consider adding graphql server here as well.
+pub(crate) async fn setup_indexer_backed_rpc(
```

Comment:
This should simply be the `new` function of `IndexerHandle`. It will make this function a lot more discoverable.